lib/livebook/file_system/s3/client.ex

```diff
@@ -304,6 +311,28 @@ defmodule Livebook.FileSystem.S3.Client do
 
   defp decode({:error, _} = error), do: error
 
+  defp ensure_credentials(%S3{access_key_id: nil} = file_system) do
```

Comment:
I believe we want to move this function somewhere up. That's because this is not the only place where we access the file system. For example, [`Kino.FS.file_spec/1`](https://hexdocs.pm/kino/Kino.FS.html#file_spec/1) allows you to access it too and I think we should have the credentials set there too.

@jonatanklosko is there a place where it would be best to add this functionality? Or should we expose `ensure_credentials` somehow through the FileSystem protocol?

Reply:
We also need to expose the secret token via Kino.FS.

Reply:
Are the current configured/saved credentials exposed? Unless I'm reading something wrong, it looks like all of the list/get/put logic is in the S3 FileSystem implementation, and that Kino.FS calls back to the runtime via `io_request` to access whatever it needs to.

Reply:
Kino.FS.file_spec allows you to retrieve the credentials so you can pass them to Explorer and other projects which may want to talk to S3 directly. So whenever that happens, we need to make sure the secrets are there.

Reply:
I think we shouldn't put the fetched credentials in the struct at any point, just so we don't persist them anywhere by accident. So the idea is that instead of `ensure_credentials` we can have `FileSystem.S3.credentials(file_system) => %{access_key_id: String.t() | nil, secret_access_key: String.t() | nil, session_token: String.t() | nil}`. Then whenever we need to use credentials (before request, and for `Kino.FS.file_spec`), we call that function instead of accessing the struct fields. Note that in this approach we also don't need the `:session_token` field on the struct altogether. @josevalim wdyt?

Reply:
Good call @jonatanklosko, I like your approach!

Reply:
I'll update the alternate branch with this change and merge it into this PR branch.